### ISSUE TITLE
Re-train all PlantSeg models to eliminate hallucination during inference

### ISSUE THREAD

**Issue by qin-yu** [hci-unihd/plant-seg#224]
## Re-train all PlantSeg models to eliminate hallucination during inference

We have historically addressed the tiling artifact by using larger patch sizes. Recently, @Buglakova re-highlighted this issue in [Issue #190](https://github.com/hci-unihd/plant-seg/issues/190), prompting further investigation in the PlantSeg and pytorch-3dunet projects. See the related [PR #220 for PlantSeg](https://github.com/hci-unihd/plant-seg/pull/220) and [PR #113 for pytorch-3dunet](https://github.com/wolny/pytorch-3dunet/pull/113).

**Findings:**
After adjusting the halo implementation, it's clear that nuclear models trained with batch normalization (with the default `track_running_stats=True`) avoid prediction hallucinations on new datasets, unlike those trained with group normalization.

**Previous Assumptions:**
We assumed group norm performed better under certain conditions based on earlier hyperparameter tuning with datasets from similar distributions. However, this might need reevaluation as batch norm has proven more stable across diverse datasets.

**Action Required:**
I propose updating versions for models previously trained with group norm. This is critical to ensure consistency and reliability in our semantic segmentation tasks.

**Visual Evidence:**
- **Left:** Prediction from a model trained with group norm using an ovules nuclear dataset.
- **Right:** Prediction from a model trained with batch norm using an ovules nuclear dataset.
- **Middle:** Raw image of a mouse embryo (Note: Brightness is increased to highlight the absence of signal).

![Comparison of model predictions](https://github.com/hci-unihd/plant-seg/assets/17722010/9cac2fde-2ad8-42be-8def-de8ff6e52394)

@wolny, could you assist in identifying the training datasets used for each model to facilitate these updates? I'll do the training after my CBB Seminar talk tomorrow.

**Comment by KRiedmiller** [comment]
I've tested some of the built-in models, and they have not yet been retrained. The tiling becomes visible only when increasing exposure a lot, but I'll evaluate all models again. If we retrain the models, this would be a good opportunity to upload them properly to bioimageIO with matching metadata compared to Panseg. 

## TODO

* [ ] figure out how patch size is defined, how it interacts with the halo, and whether the automatic patch size determination makes sense or should be left at the training patch size. (Also sometimes OOM: #578)
* [ ] First tests suggest small patch sizes (16,64,64) together with a bigger batch number works better than the other way around. Previous models were trained with patch size of (80,160,160), which makes it hard to fit many batches into VRAM.  -> Find good patch size
* [ ] Gather list of datasets of built-in models. List of built-in models: https://github.com/kreshuklab/panseg/issues/551#issuecomment-4543962177
* [ ] Make sure models trained in Panseg are as compatible with BioimageIO as possible, as the newly trained models should get uploaded there, and build into panseg. (Once feature parity is achieved we can get rid of our own inference code) #551 
* [ ] Retrain models. Easy.
* [ ] Update Built-in models
* [ ] Update BioimageIO models. (new versions? new unrelated models?)

**Comment by KRiedmiller** [comment]
See #577 for Panseg built-in training. With this PR, the training uses batchnorm.
